proc/proc.go

```diff
@@ -175,3 +176,48 @@ func FindProcessByName(name string) (*os.Process, error) {
 
 	return nil, fmt.Errorf("process %s is not found", name)
 }
+
+// FindProcessByCmdline finds the processes with matching cmdline
+func FindProcessByCmdline(cmdline string) ([]*os.Process, error) {
+	processes := []*os.Process{}
+
+	procDir, err := os.Open("/proc")
+	if err != nil {
+		return nil, err
+	}
+	defer procDir.Close()
+
+	ps, err := procDir.Readdir(-1)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, p := range ps {
+		if p.IsDir() {
+			pid, convErr := strconv.Atoi(p.Name())
+			if convErr != nil {
+				continue
+			}
```

Comment:
Have you thought about reusing the code from `go-ps` lib? So don't need to do parsing by yourself. 

```go
	processes, err := ps.Processes()
	if err != nil {
		return nil, fmt.Errorf("failed to list processes")
	}

	for _, process := range processes {
        // TODO your implementation ..
	}
```

Reply:
Yes, I thought the idea, but go-ps gets process name from `/proc/<pid>/status` rather than `/proc/<pid>/cmdline`.

Reply:
I see, but what I meant is you can use the lib to do below instead, so you don't need to scan the proc directory again to get all processes, just reuse the lib. WDYT? 

```go
func FindProcessByCmdline(cmdline string) ([]*os.Process, error) {
	processes, err := ps.Processes()
	if err != nil {
		return nil, fmt.Errorf("failed to list processes")
	}

	var foundProcesses []*os.Process
	
	for _, process := range processes {
		cmdlinePath := filepath.Join("/proc", strconv.Itoa(process.Pid()), "cmdline")
		cmdlineContent, readErr := os.ReadFile(cmdlinePath)
		if readErr != nil {
			continue
		}

		if strings.HasPrefix(string(cmdlineContent), cmdline) {
			process, err := os.FindProcess(process.Pid())
			if err != nil {
				return nil, err
			}
			foundProcesses = append(foundProcesses, process)
		}
	}
	
	if len(foundProcesses) == 0 {
		return nil, fmt.Errorf("process with cmdline %s is not found", cmdline)
	}

	return foundProcesses, nil
}
```

Reply:
Ah, misunderstood. Addressed, thank you.